gradle-plugin/src/main/groovy/grails/views/gradle/util/SourceSets.groovy

```diff
@@ -0,0 +1,46 @@
+package grails.views.gradle.util
+
+import groovy.transform.CompileStatic
+import org.gradle.api.Project
+import org.gradle.api.plugins.JavaPluginExtension
+import org.gradle.api.tasks.SourceSet
+import org.gradle.api.tasks.SourceSetContainer
+
+/**
+ * Copied from grails-gradle-plugin to remove the dependency on that plugin.
+ * This is a possible candidate for a future "grails-common" module.
+ *
+ * @author Graeme Rocher
+ * @since 3.0
+ */
+@CompileStatic
+class SourceSets {
```

Comment:
And this file

Reply:
https://github.com/grails/grails-core/issues/13850#issuecomment-2558122348